spec/lexers/ecl_spec.rb

```diff
@@ -0,0 +1,27 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+describe Rouge::Lexers::ECL do
+  let(:subject) { Rouge::Lexers::ECL.new }
+
+  describe 'guessing' do
+    include Support::Guessing
+
+    it 'guesses by filename' do
+      assert_guess :filename => 'foo.ecl'
+      assert_guess :filename => 'FOO.ECL'
```

Comment:
You don't need to check this.

Reply:
@pyrmont I'm not having luck matching everything between words for our "embed" function. The regex doesn't do that from what I can see. Is there a way to **mixin** languages like Python, c++, etc? This doesn't work. It matches to the last ENDEMBED in the file:

`EMBED[\s\S]*ENDEMBED;`

Our syntax is:

```
INTEGER addone(INTEGER p) := EMBED(Python)
# Python code that returns one more than the value passed to it\n
if p < 10:\n
    return p+1\n
else:\n
    return 0\n
ENDEMBED;
```

Suggestions?

Reply:
What you probably want is the `delegate` function. And in general, when writing regexes in a lexer, you want every `*` or `+` to be followed by a `?` - this makes it non-greedy and will match the shortest possible string.

Reply:
See for example the HTML lexer, which embeds both CSS and Javascript, for an example of `delegate`.

Reply:
Thanks!

Reply:
Are those `\n`s actually in the source, or is that just an indicator that there are newlines there?

Reply:
Sorry, shouldn't be there. Fixed.

Reply:
Just a reminder that you don't need to check the different cases. This line can be deleted.